### PR TITLE
image: Build an image from cmd

### PR DESCRIPTION
**Pull request by orelmisan**
This adds a Dockerfile and a simple build system for the image:
```bash
./automation/make.sh --build-core-image
```

It supports podman as a default, with the ability to choose arbitrary CRIs such as Docker:
```bash
CRI=docker ./automation/make.sh --build-core-image
```

An image tag can also be set:
```bash
CORE_IMAGE_TAG=0.1.0 ./automation/make.sh --build-core-image
```

~~This PR depends on PR #21.~~

### CODE REVIEW

File: automation/make.sh [hunk 1]
```diff
@@ -31,6 +37,9 @@ while true; do
     --unit-test)
         OPT_UNIT_TEST=1
         ;;
+    --build-image)
```
Comment (EdDev) [inline automation/make.sh:40]: We need to differentiate between the "core" and the checkups.
Maybe `build-core-image` ?

File: automation/make.sh [hunk 2]
```diff
@@ -31,6 +37,9 @@ while true; do
...
+        BUILD_IMAGE=1
+        ;;
     --help)
         set +x
         echo "$0 [--lint] [--unit-test]"
```
Comment (EdDev) [inline automation/make.sh:45]: You should add it to the help as well.
And to the "default" section condition. But I would not enable the option if not set explicitly.

File: automation/make.sh [hunk 3]
```diff
@@ -31,6 +37,9 @@ while true; do
     --unit-test)
         OPT_UNIT_TEST=1
         ;;
+    --build-image)
+        BUILD_IMAGE=1
```
Comment (EdDev) [inline automation/make.sh:41]: Please prefix with `OPT_`.

File: automation/make.sh [hunk 4]
```diff
@@ -19,8 +19,14 @@
 
 set -e
 
+CRI=${CRI:-docker}
+IMAGE_TAG=${IMAGE_TAG:-latest}
+
+BINARY_NAME="kiagnose"
+IMAGE_NAME="kiagnose"
```
Comment (EdDev) [inline automation/make.sh:26]: `CORE` needs to be specified here so we can differentiate between this and the individual checkup references.

File: automation/make.sh [hunk 5]
```diff
@@ -19,8 +19,14 @@
 
 set -e
 
+CRI=${CRI:-docker}
+IMAGE_TAG=${IMAGE_TAG:-latest}
```
Comment (EdDev) [inline automation/make.sh:23]: Lets not use `latest` as it causes some less preferable defaults on a k8s cluster [1].
How about `debug` or `devel`?

[1] https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/f58fcfd2725fc8d6d3d8cdf903577a1b43c3a7da..7208a49fb1f1beb87c63f55b323e428fcacb4a7a): Rebase and fixed a code review comment.

File: automation/make.sh (hunk 3 above)
Comment (orelmisan) [reply automation/make.sh]: Done.

File: automation/make.sh (hunk 2 above)
Comment (orelmisan) [reply automation/make.sh]: Done.

File: automation/make.sh (hunk 5 above)
Comment (orelmisan) [reply automation/make.sh]: Used `devel`.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/7208a49fb1f1beb87c63f55b323e428fcacb4a7a..149065fe06feaeaf951e7b39fece97cd791f0f9d): Fixed code review comments.

File: automation/make.sh (hunk 1 above)
Comment (orelmisan) [reply automation/make.sh]: Done.

File: automation/make.sh [hunk 6]
```diff
@@ -19,10 +19,14 @@
 
 set -e
 
+CRI=${CRI:-docker}
```
Comment (EdDev) [inline automation/make.sh:22]: Can we make it `podman`?

**Review by EdDev — COMMENTED** [review]
Can you place the documentation on how to use it in the commit message as well?
All good except that small part.

File: automation/make.sh (hunk 6 above)
Comment (orelmisan) [reply automation/make.sh]: Done.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/149065fe06feaeaf951e7b39fece97cd791f0f9d..509eb81aa1171496c868b8194f9489f14412af4e): Made `podman` the default container engine.

**Review by EdDev — DISMISSED** [review]
Thank you!

**Comment by EdDev** [comment]
Please rebase

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/509eb81aa1171496c868b8194f9489f14412af4e..103680228fac7c4a62c9c9dcdb4a45b698d7e3ee): Rebase.

**Review by EdDev — COMMENTED** [review]
Sorry, I missed one thing: The dockerfile and explicitly pointing the build command to it.
Lets place it in a folder, e.g.  `containerfile` / `dockerfile` / `imagespec`  / `packaging` and name it with a posix that mentions it is "kiagnose". Later we will also have the ref checkups there.

EDIT:
`Dockerfile.kiagnose`

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/103680228fac7c4a62c9c9dcdb4a45b698d7e3ee..1092f9f2befc673f0bb8751713dfb13a34311478): Moved `Dockerfile` to `dockerfiles/Dockerfile.kiagnose`

**Review by EdDev — APPROVED** [review]
Thanks!

**Pull request merged**